### ISSUE TITLE
JIT SVE: Assertion failed '!m_VariableLiveRanges->back().m_EndEmitLocation.Valid()' during 'Generate code'

### ISSUE THREAD

**Issue by jakobbotsch** [dotnet/runtime#105723]
```csharp
// Generated by Fuzzlyn v2.2 on 2024-07-30 18:12:19
// Run on Arm64 Windows
// Seed: 59622729874143167-vectort,vector64,vector128,armsve
// Reduced from 40.7 KiB to 0.7 KiB in 00:00:36
// Hits JIT assert in Release:
// Assertion failed '!m_VariableLiveRanges->back().m_EndEmitLocation.Valid()' in 'Program:Main(Fuzzlyn.ExecutionServer.IRuntime)' during 'Generate code' (IL size 71; hash 0xade6b36b; FullOpts)
//
//     File: C:\dev\dotnet\runtime\src\coreclr\jit\scopeinfo.cpp Line: 864
//
using System;
using System.Runtime.CompilerServices;
using System.Numerics;
using System.Runtime.Intrinsics;
using System.Runtime.Intrinsics.Arm;

public class Program
{
    public static IRuntime s_rt;
    public static void Main()
    {
        Vector<long> vr2 = Vector128.CreateScalar(-2L).AsVector();
        var vr3 = new sbyte[]
        {
            1
        };
        var vr4 = Vector128.CreateScalar(9223372036854775806L).AsVector();
        Vector<long> vr5 = Sve.MultiplySubtract(vr2, vr2, vr4);
        s_rt.WriteLine(vr5);
    }
}

public interface IRuntime
{
    void WriteLine<T>(T value);
}

public class Runtime : IRuntime
{
    public void WriteLine<T>(T value) => System.Console.WriteLine(value);
}
```

cc @dotnet/jit-contrib @dotnet/arm64-contrib 

This is a common symptom of missing or incorrect liveness updating (`genConsumeReg`/`genProduceReg`), presumably related to `MultiplySubtract`, so that's the first place I would suggest to look.

**Comment by a74nh** [comment]
priority:2 for RC1 snap : Functionality issue

**Comment by a74nh** [comment]
```
Scope info: begin block BB01, IL range [000..042)
Added IP mapping: 0x0000 STACK_EMPTY (G_M7095_IG02,ins#0,ofs#0) label
Generating: N003 (???,???) [000062] -----------                            IL_OFFSET void   INLRT @ 0x000[E-] REG NA
Generating: N005 (  3,  2) [000002] -----------                    t2 =    CNS_VEC   simd16<0xfffffffe, 0xffffffff, 0x00000000, 0x00000000> REG d0 $140
Increasing data section alignment from 4 to 16 for type simd16
Mapped BB01 to G_M7095_IG02
IN0001:             ldr     q0, (LARGELDC)[@RWD00]
                                                                        /--*  t2     simd16 
Generating: N007 (  3,  3) [000003] DA---------                         *  STORE_LCL_VAR simd16<System.Numerics.Vector`1> V00 loc0         d:2 NA REG NA $VN.Void
IN0002:             str     q0, [fp, #0x20]	// [V00 loc0]
Debug: New V00 debug range: first
							Live vars after [000003]: {} +{V00} => {V00}
Generating: N009 (  3, 12) [000050] -----------                   t50 =    CNS_INT   long   0xea5eddf146b0 REG x19 $184
IN0003:             movz    x19, #0x46B0
IN0004:             movk    x19, #0xDDF1 LSL #16
IN0005:             movk    x19, #0xEA5E LSL #32
                                                                        /--*  t50    long   
Generating: N011 (  3, 12) [000051] DA---------                         *  STORE_LCL_VAR long   V08 cse0         d:1 x19 REG x19 $VN.Void
							V08 in reg x19 is becoming live  [000051]
							Live regs: 0000000000000000 {} + {x19} => 0000000000080000 {x19}
							Live vars after [000051]: {V00} +{V08} => {V00 V08}
Generating: N013 (  1,  1) [000052] -----------                   t52 =    LCL_VAR   long   V08 cse0         u:1 x19 REG x19 $184
Generating: N015 (  1,  2) [000053] -c---------                   t53 =    CNS_INT   long   648 REG NA
                                                                        /--*  t52    long   
                                                                        +--*  t53    long   
Generating: N017 (  3,  4) [000054] -------N---                   t54 = *  ADD       long   REG x0 $180
IN0006:             add     x0, x19, #648
                                                                        /--*  t54    long   
Generating: N019 (???,???) [000065] -----------                   t65 = *  PUTARG_REG long   REG x0
Generating: N021 (  1,  2) [000004] -----------                    t4 =    CNS_INT   long   1 REG x1 $1c0
IN0007:             mov     x1, #1
                                                                        /--*  t4     long   
Generating: N023 (???,???) [000066] -----------                   t66 = *  PUTARG_REG long   REG x1
                                                                        /--*  t65    long   arg0 in x0
                                                                        +--*  t66    long   arg1 in x1
Generating: N025 ( 21, 22) [000006] -ACXG------                    t6 = *  CALL help ref    CORINFO_HELP_NEWARR_1_VC REG x0 $242
Call: GCvars=0000000000000000 {}, gcrefRegs=0000 {}, byrefRegs=0000 {}
IN0008:             bl      CORINFO_HELP_NEWARR_1_VC
							GC regs: 0000 {} => 0001 {x0}
                                                                        /--*  t6     ref    
Generating: N027 ( 21, 22) [000035] DACXG------                         *  STORE_LCL_VAR ref    V07 tmp4         d:2 x0 REG x0 $243
							GC regs: 0001 {x0} => 0000 {}
							V07 in reg x0 is becoming live  [000035]
							Live regs: 0000000000080000 {x19} + {x0} => 0000000000080001 {x0 x19}
							Live vars after [000035]: {V00 V08} +{V07} => {V00 V07 V08}
							GC regs: 0000 {} => 0001 {x0}
Generating: N029 (  1,  1) [000037] -----------                   t37 =    LCL_VAR   ref    V07 tmp4         u:2 x0 (last use) REG x0 $280
                                                                        /--*  t37    ref    
Generating: N031 (  1,  1) [000042] -c---------                   t42 = *  LEA(b+16) byref  REG NA
Generating: N033 (  1,  2) [000011] -----------                   t11 =    CNS_INT   int    1 REG x1 $44
IN0009:             mov     w1, #1
                                                                        /--*  t42    byref  
                                                                        +--*  t11    int    
Generating: N035 ( 28, 29) [000013] nACXGO-----                         *  STOREIND  byte   REG NA $243
							V07 in reg x0 is becoming dead  [000037]
							Live regs: 0000000000080001 {x0 x19} - {x0} => 0000000000080000 {x19}
							Live vars after [000037]: {V00 V07 V08} -{V07} => {V00 V08}
							GC regs: 0001 {x0} => 0000 {}
IN000a:             strb    w1, [x0, #0x10]
Added IP mapping: 0x002D STACK_EMPTY (G_M7095_IG02,ins#10,ofs#48)
Generating: N037 (???,???) [000063] -----------                            IL_OFFSET void   INLRT @ 0x02D[E-] REG NA
Generating: N039 (  1,  1) [000017] -----------                   t17 =    LCL_VAR   simd16<System.Numerics.Vector`1> V00 loc0         u:2 d0 (last use) REG d0 $140
Generating: N041 (  1,  1) [000018] ----------z                   t18 =    LCL_VAR   simd16<System.Numerics.Vector`1> V00 loc0         u:2 d0 REG d0 $140
Generating: N043 (  3,  2) [000015] -----------                   t15 =    CNS_VEC   simd16<0xfffffffe, 0x7fffffff, 0x00000000, 0x00000000> REG d16 $141
IN000b:             ldr     q16, (LARGELDC)[@RWD16]
Generating: N045 (???,???) [000067] -----------                   t67 =    HWINTRINSIC mask   long CreateTrueMaskAll REG p0
IN000c:             ptrue   p0.d
Generating: N047 (???,???) [000068] -c---------                   t68 =    CNS_VEC   simd16<0x00000000, 0x00000000, 0x00000000, 0x00000000> REG NA
                                                                        /--*  t17    simd16 
                                                                        +--*  t18    simd16 
                                                                        +--*  t15    simd16 
Generating: N049 (  6,  5) [000020] -c---------                   t20 = *  HWINTRINSIC simd16 long MultiplySubtract REG NA $400
                                                                        /--*  t67    mask   
                                                                        +--*  t20    simd16 
                                                                        +--*  t68    simd16 
Generating: N051 (???,???) [000069] -----------                   t69 = *  HWINTRINSIC simd16 long ConditionalSelect REG d0
Debug: Closing V00 debug range.
							Live vars after [000017]: {V00 V08} -{V00} => {V08}
IN000d:             ldr     q0, [fp, #0x20]	// [V00 loc0]

Assert failure(PID 3098694 [0x002f4846], Thread: 3098694 [0x2f4846]): Assertion failed '!m_VariableLiveRanges->back().m_EndEmitLocation.Valid()' in 'Program:myTest()' during 'Generate code' (IL size 66; hash 0xb943e448; FullOpts)

    File: /home/alahay01/dotnet/runtime_sve/src/coreclr/jit/scopeinfo.cpp:864
    Image: /home/alahay01/dotnet/runtime_sve/artifacts/tests/coreclr/linux.arm64.Checked/Tests/Core_Root/corerun

[1]    3098694 abort (core dumped)  $CORE_ROOT/corerun  > ~/dump_live.txt
```

Consuming the registers for `MultiplySubtract`. `t17` is consumed ok, and then `t18` is consumed which fails.

`t17` and `t18` and the same variable at C# level. Oddly `t17` is marked as last use, but `t18` isn't

**Comment by jakobbotsch** [comment]
It seems to suggest that the uses of `MultiplySubtract` are being built in the wrong order by LSRA. The issue may be similar in spirit to #103100.

**Comment by a74nh** [comment]
> wrong order by LSRA

... yes, it looks like LSRA is swapping the operands:


before:
```
N001 (  1,  1) [000017] -----------                   t17 =    LCL_VAR   simd16<System.Numerics.Vector`1> V00 loc0         u:2 $140
N002 (  1,  1) [000018] -----------                   t18 =    LCL_VAR   simd16<System.Numerics.Vector`1> V00 loc0         u:2 (last use) $140
```

lsraarm64.cpp:
```
            unsigned resultOpNum =
                embOp2Node->GetResultOpNumForRmwIntrinsic(user, intrinEmb.op1, intrinEmb.op2, intrinEmb.op3);

            GenTree* emitOp1 = intrinEmb.op1;
            GenTree* emitOp2 = intrinEmb.op2;
            GenTree* emitOp3 = intrinEmb.op3;

            if (resultOpNum == 2)
            {
                // op2 = op1 + (op2 * op3)
                std::swap(emitOp1, emitOp3);
                std::swap(emitOp1, emitOp2);
                // op1 = (op1 * op2) + op3
            }
```

after:
```
N039 (  1,  1) [000017] -----------                   t17 =    LCL_VAR   simd16<System.Numerics.Vector`1> V00 loc0         u:2 d0 (last use) REG d0 $140
N041 (  1,  1) [000018] ----------z                   t18 =    LCL_VAR   simd16<System.Numerics.Vector`1> V00 loc0         u:2 d0 REG d0 $140
```

**Comment by a74nh** [comment]
It seems the swaps happen assuming they will also be swapped during codegen. But the `genConsumeMultiOpOperands()` (where the failure happens) is before the codegen swapping.

**Comment by a74nh** [comment]
Deleting the logic in lsraarm64.cpp "fixes" the issue, but I need to figure out if that is safe....

**Comment by jakobbotsch** [comment]
The simple thing to do is to just write it similar to what I did in the fix in #103100 -- as a loop that ensures the uses are built in the right order, but that retains the same preferencing that is there right now.
